src/Core/ShellUtility.re

```diff
@@ -3,27 +3,66 @@ open Kernel;
 module Log = (val Log.withNamespace("Oni2.Core.ShellUtility"));
 
 module Internal = {
-  let getDefaultShell = () => {
-    switch (Sys.getenv_opt("SHELL")) {
-    | Some(v) => v
-    | None => Sys.win32 ? "cmd.exe" : "/usr/bin/bash"
-    };
-  };
-
-  let getShellFromPath = () =>
+  let getPathFromEnvironment = () =>
     switch (Sys.getenv_opt("PATH")) {
     | Some(path) => path
     | None =>
       Log.error("Unable to get PATH!");
       "";
     };
+
+  // This strategy for determing the default shell came from StackOverflow:
+  // https://stackoverflow.com/a/41553295
+  // This is important because in some cases, like launching from Finder,
+  // there may not be an $SHELL environment variable for us.
+  let discoverOSXShell = () =>
+    try({
+      let (stdOut, stdIn, stdErr) =
+        Unix.open_process_full("/usr/bin/dscl . read ~/ UserShell", [||]);
+      let userShell = input_line(stdOut);
+      let () = close_in(stdOut);
+      let () = close_out(stdIn);
+      let () = close_in(stdErr);
+      // Returns a string of the form: "UserShell: /bin/zsh"
+
+      let len = String.length(userShell);
+      let slashIndex = String.index(userShell, '/');
+      Log.infof(m => m("dscl returned: %s", userShell));
+
+      String.sub(userShell, slashIndex, len - slashIndex);
+    }) {
+    | ex =>
+      Log.warn("Unable to run dscl to get user shell");
+      "/bin/bash";
```

Comment:
Isn't `zsh` the default shell on macOS now?

Reply:
Thanks, it is. I've updated the default to use `zsh`